### PR TITLE
Clear out submodules and restructure slightly

### PR DESCRIPTION
**Pull request by tfoote**
@sloretz This is a slightly cleaned up copy of what we looked at with the cache merged into the deploy.

### CODE REVIEW

File: resources.repos [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+    type: git
+    url: git@github.com:ros/rosdistro.git
+    version: master
+  rosforks:
+    type: git
+    url: git@github.com:rosindex/rosforks.git
+    version: master
+  deploy:
+    type: git
+    url: git@github.com:rosindex/rosindex.github.io.git
```
Comment (sloretz) [inline resources.repos]: `ros-infrastructure/rosindex`?

File: _config.yml [hunk 2]
```diff
@@ -32,11 +32,11 @@ redcarpet:
 
 # If true, this begins with an existing db
 use_db_cache: false
-db_cache_filename: _caches/_deploy_cache.db
-report_filename: _caches/_deploy_index_report.yaml
-report_diff_filename: _caches/_deploy_index_report_diff.yaml
-debbuild_release_path: _caches/debbuild_releases
-wiki_title_index_filename: _caches/ros_org_wiki_title_index.txt
+db_cache_filename: ../deploy/cache/_deploy_cache.db
```
Comment (sloretz) [inline _config.yml]: `checkout.sh` doesn't create the `cache` directory. Maybe touch a .gitignore file in the folder so it gets added in this PR?

Comment (sloretz) [reply _config.yml]: I rebased #7 on top of this PR and made it make the cache directory in the `build` rule.

File: resources.repos (hunk 1 above)
Comment (hidmic) [reply resources.repos]: +1 I don't think we'd want to deploy against public rosindex, though I can't seem to find said repository in `ros-infrastructure` :/

**Comment by apojomovsky** [comment]
Just tested this along to the latest changes from [rosindex_containers#1](https://github.com/ros-infrastructure/rosindex_containers/pull/1) and is working as expected.
The rosindex submodule found in there are being updated to reflect this merge.

**Pull request merged**